File: 0389_find_the_difference.swift
// https://leetcode.com/problems/find-the-difference/

class Solution {
	func findTheDifference(_ s: String, _ t: String) -> Character {
		var dic = Dictionary<Character, Int>()
		for c in s {
			dic[c] = (dic[c] ?? 0) + 1
		}
		for c in t {
			if let count = dic[c] {
				if count == 1 {
					dic.removeValue(forKey: c)
				} else {
					dic[c] = count - 1
				}
			} else {
				return c
			}
		}
		return "."
	}
}

File: 0389_find_the_difference.kt
// https://leetcode.com/problems/find-the-difference/description/

class Solution {
    fun findTheDifference(s: String, t: String): Char {
        val sortedS = s.toCharArray().sorted()
        val sortedT = t.toCharArray().sorted()
        for (i in sortedS.indices) {
            if (sortedS[i] != sortedT[i]) {
                return sortedT[i]
            }
        }
        return sortedT[sortedT.size - 1]
    }
}
